src/System.Net.Security/tests/FunctionalTests/ServiceNameCollectionTest.cs

```diff
@@ -0,0 +1,260 @@
+            int[] destination = new int[collection.Count];
+
+            // The full framework throws InvalidCastException.
+            Assert.Throws<ArgumentException>(() => collection.CopyTo(destination, 0));
+        }
+
+        [Fact]
+        public void Enumerator_BehavesAsExpected()
+        {
+            string item1 = "first";
+            string item2 = "second";
+            string item3 = "third";
+
+            var collection = new ServiceNameCollection(new[] { item1, item2, item3 });
+
+            IEnumerator e = collection.GetEnumerator();
+
+            for (int i = 0; i < 2; i++)
+            {
+                // Not started
+                Assert.Throws<InvalidOperationException>(() => e.Current);
+
+                Assert.True(e.MoveNext());
+                Assert.Same(item1, e.Current);
+
+                Assert.True(e.MoveNext());
+                Assert.Same(item2, e.Current);
+
+                Assert.True(e.MoveNext());
+                Assert.Same(item3, e.Current);
+
+                Assert.False(e.MoveNext());
+                Assert.False(e.MoveNext());
+                Assert.False(e.MoveNext());
+
+                // Ended
+                Assert.Throws<InvalidOperationException>(() => e.Current);
+
+                e.Reset();
+            }
+        }
+
+        [Fact]
+        public void Merge_NullOrEmptyString_Throws()
+        {
+            var collection = new ServiceNameCollection(new[] { "first", "second" });
+            Assert.Throws<ArgumentException>(() => collection.Merge((string)null));
+            Assert.Throws<ArgumentException>(() => collection.Merge(string.Empty));
+        }
+
+        [Fact]
+        public void Merge_NonDuplicateString_Succeeds()
+        {
+            string[] expected = new[] { "first", "second", "third" };
+            var collection = new ServiceNameCollection(new[] { "first", "second" });
+
+            ServiceNameCollection merged = collection.Merge("third");
+            Assert.Equal(expected, merged.Cast<string>());
+        }
+
+        [Theory]
+        [InlineData("first")]
+        [InlineData("SECOND")]
+        public void Merge_DuplicateString_DuplicateFiltered(string duplicate)
+        {
+            string[] expected = new[] { "first", "second" };
+            var collection = new ServiceNameCollection(new[] { "first", "second" });
+
+            ServiceNameCollection merged = collection.Merge(duplicate);
+            Assert.Equal(expected, merged.Cast<string>());
+        }
+
+        [Fact]
+        public void Merge_NullEnumerable_Throws()
+        {
+            var collection = new ServiceNameCollection(new[] { "first", "second" });
+
+            // This really should be ArgumentNullException, but the full framework does not
+            // guard against null, so NullReferenceException ends up being thrown instead.
+            Assert.Throws<NullReferenceException>(() => collection.Merge((IEnumerable)null));
```

Comment:
You mean changing the product code?  If so, don't do that since that will impact app-compat.